src/include/concurrency/transaction_state_handler.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "transaction_manager_factory.h"
+
+namespace peloton {
+namespace concurrency {
+  enum class TransactionState{
+   IDLE = 0,
+   IMPLICIT_START,
+    EXPLICIT_START,
+    ABORTING,
+    ENDED,
+ };
+
+class TransactionStateHandler {
```

Comment:
Maybe instead of TransactionStateHandler, call it ClientTransactionHandle or something like that, since this component stores all relevant information in the network layer about what a client's transaction is doing, instead of performing anything on transactions.